pkg/kudoctl/cmd/diagnostics/runner_helper.go

```diff
@@ -0,0 +1,157 @@
+package diagnostics
+
+import (
+	"github.com/kudobuilder/kudo/pkg/kudoctl/env"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	"github.com/kudobuilder/kudo/pkg/version"
+)
+
+type runnerHelper struct {
+	p *nonFailingPrinter
+}
+
+func (rh *runnerHelper) runForInstance(instance string, options *Options, c *kudo.Client, info version.Info, s *env.Settings) error {
+	ir, err := newInstanceResources(instance, options, c, s)
+	if err != nil {
+		return err
+	}
+
+	ctx := &processingContext{root: DiagDir, instanceName: instance}
+	instanceDiagRunner := &runner{}
+	instanceDiagRunner.
+		run(resourceCollectorGroup{
+			{
+				loadResourceFn: ir.instance,
+				errKind:        "instance",
+				parentDir:      ctx.attachToOperator,
+				failOnError:    true,
+				callback:       ctx.mustSetOperatorVersionNameFromInstance,
+				printer:        rh.p,
+				printMode:      ObjectWithDir},
+			{
+				loadResourceFn: ir.operatorVersion(ctx.operatorVersionName),
+				errKind:        "operatorversion",
+				parentDir:      ctx.attachToOperator,
+				failOnError:    true,
+				callback:       ctx.mustSetOperatorNameFromOperatorVersion,
+				printer:        rh.p,
+				printMode:      ObjectWithDir},
+			{
+				loadResourceFn: ir.operator(ctx.operatorName),
+				errKind:        "operator",
+				parentDir:      ctx.attachToRoot,
+				failOnError:    true,
+				printer:        rh.p,
+				printMode:      ObjectWithDir}}).
+		run(&resourceCollector{
+			loadResourceFn: ir.pods,
+			errKind:        "pod",
+			parentDir:      ctx.attachToInstance,
+			callback:       ctx.mustAddPodNames,
+			printer:        rh.p,
+			printMode:      ObjectListWithDirs}).
+		run(&resourceCollector{
+			loadResourceFn: ir.services,
+			errKind:        "service",
+			parentDir:      ctx.attachToInstance,
+			printer:        rh.p,
+			printMode:      RuntimeObject}).
```

Comment:
Not a big fan of the builder pattern here. What's the upside of

```
r := &runner{}
r.run(...).run(...).run(...)
```
vs
```
r := &runner{}
r.run(...)
r.run(...)
r.run(...)
```
?

I can parse the latter one a lot easier, but that might be personal preference, so not a blocker here.

Reply:
@ANeumann82 I agree it's totally a matter of preference. I perhaps met too many people who like chaining :) As a very shaky argument, perhaps falling through in case of error looks slightly more natural for a chain.
@nfnt I guess it's also a matter of aesthetic preferences here. Imho, the amount of added repetitive boilerplate code here is an example of what so many people hate about Go's error handling. I think if efforts are put into uniform collectors, it makes sense to benefit from them and hide boilerplate into a primitive runner.